packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -406,6 +410,31 @@ export async function manifestsMigration(context: MigrationContext): Promise<voi
     );
   }
 
+  const oldManifestPath = path.join(oldAppPackageFolderPath, MANIFEST_TEMPLATE_CONSOLIDATE);
+  const oldManifestExists = await fs.pathExists(path.join(context.projectPath, oldManifestPath));
+  if (!oldManifestExists) {
+    // templates/appPackage/manifest.template.json does not exist
+    throw MigrationError(
+      new Error(getLocalizedString("core.migrationV3.manifestNotExist")),
```

Comment:
is it possible to remove line 403-411 and throw this error istead? Seems we only care whether the manifest file exists, not whether the app package folder exists?